solr/modules/sql/src/java/org/apache/solr/handler/sql/SolrFilter.java

```diff
@@ -356,13 +367,16 @@ protected String translateLike(RexNode like) {
       terms = translateLikeTermToSolrSyntax(terms, escapeChar);
 
       if (!terms.startsWith("(") && !terms.startsWith("[") && !terms.startsWith("{")) {
-        terms = escapeWithWildcard(terms);
+        String solrFieldType = solrFieldTypes.get(pair.getKey()).getType();
+        terms = escapeWithWildcard(terms, solrFieldType);
 
         // if terms contains multiple words and one or more wildcard chars, then we need to employ
         // the complexphrase parser
         // but that expects the terms wrapped in double-quotes, not parens
         boolean hasMultipleTerms = terms.split("\\s+").length > 1;
-        if (hasMultipleTerms && (terms.contains("*") || terms.contains("?"))) {
+        if (solrFieldType.equals("text")
+            && hasMultipleTerms
```

Comment:
Let's think about the order we want these checks, I think there primitive Boolean probably should be first. And then reverse the check to be "text".equals to be better about null handling.